src/osd/PG.cc

```diff
@@ -1380,11 +1380,18 @@ Scrub::schedule_result_t PG::sched_scrub()
 
   // try to reserve the local OSD resources. If failing: no harm. We will
   // be retried by the OSD later on.
-  if (!m_scrubber->reserve_local()) {
-    dout(10) << __func__ << ": failed to reserve locally" << dendl;
-    return schedule_result_t::osd_wide_failure;
+  // Note that high-priority scrubs are always granted the local resources.
+  {
+    // a temporary implementation of the 'is high priority?' check. Will be
+    // replaced by the attributes of the specific scrub target in the refactored
+    // scheduling code.
+    const bool is_high_priority =
+	updated_flags->req_scrub || updated_flags->must_scrub;
+
```

Comment:
nit: not sure whether this particular line adds to the readability but the scope limiting is a good thing!